lib/pool.js

```diff
@@ -42,24 +42,33 @@ Pool.prototype.request = function (method, path, data) {
       },
       data: data ? JSON.stringify(data) : undefined
     },
-    function (err, res, body) {
-      if (err || Math.floor(res && res.statusCode / 100) !== 2) {
-        var e = new Error(body || (err && err.message))
-        e.statusCode = res && res.statusCode
-        return d.reject(e)
-      }
-      if (!body) { return d.resolve() }
-      var json = null
-      try {
-        json = JSON.parse(body)
-      }
-      catch (e) {
-        return d.reject(new Error('Invalid JSON'))
-      }
-      d.resolve(json)
-    }
+    handleResponse
```

Comment:
With the added logic, this function felt a bit too unwieldy to be anonymous and inline. I named it and pulled it down to (hopefully) aid readability.